app/models/course/assessment/question/programming.rb

```diff
@@ -5,7 +5,7 @@ class Course::Assessment::Question::Programming < ActiveRecord::Base
 
   acts_as :question, class_name: Course::Assessment::Question.name
 
-  before_save :process_new_package, if: :attachment_changed?
+  before_save :process_new_package, if: :attachment_reference_changed?
```

Comment:
Something to consider: If the attachment reference is suppose to duck-type like the Attachment, then we shouldn't need to change any of these?